src/auth/bearer/Config.cc

```diff
@@ -0,0 +1,234 @@
+
+    if (authenticateProgram)
+        wordlistDestroy(&authenticateProgram);
+
+    safe_free(bearerAuthScope);
+}
+
+bool
+Auth::Bearer::Config::dump(StoreEntry * entry, const char *name, Auth::SchemeConfig * scheme) const
+{
+    if (!Auth::SchemeConfig::dump(entry, name, scheme))
+        return false; // not configured
+
+    if (bearerAuthScope != BearerDefaultScope)
+        entry->appendf("%s bearer scope %s\n", name, bearerAuthScope);
+
+    return true;
+}
+
+Auth::Bearer::Config::Config()
+{
+    static const SBuf defaultRealm("Squid proxy-caching web server");
+    realm = defaultRealm;
+    bearerAuthScope = BearerDefaultScope;
+}
+
+Auth::Bearer::Config::~Config()
+{
+    if (bearerAuthScope != BearerDefaultScope)
+        xfree(bearerAuthScope);
+}
+
+void
+Auth::Bearer::Config::parse(Auth::SchemeConfig * scheme, int n_configured, char *param_str)
+{
+    if (strcmp(param_str, "scope") == 0) {
+        char *tmpS = nullptr;
+        parse_eol(&tmpS);
+        if (strcmp(bearerAuthScope, tmpS) != 0) {
+            debugs(3, DBG_PARSE_NOTE(2), "NOTICE: Bearer scope already set to: " << tmpS);
+            xfree(tmpS);
+        } else
+            bearerAuthScope = tmpS;
+
+    } else
+        Auth::SchemeConfig::parse(scheme, n_configured, param_str);
+}
+
+static void
+authenticateBearerStats(StoreEntry * sentry)
+{
+    bearerauthenticators->packStatsInto(sentry, "Bearer Authenticator Statistics");
+}
+
+/**
+ * Decode a Bearer [Proxy-]Auth string. Looking for an existing
+ * Auth::UserRequest structure with matching token, or create a
+ * new one if needed.
+ *
+ * Note that just returning will be treated as
+ * "cannot decode credentials". Use the message field to return
+ * a descriptive message to the user.
+ */
+Auth::UserRequest::Pointer
+Auth::Bearer::Config::decode(char const *proxy_auth, const HttpRequest *, const char *aRequestRealm)
+{
+    Auth::UserRequest::Pointer auth_user_request = dynamic_cast<Auth::UserRequest*>(new Auth::Bearer::UserRequest);
+
+    // retrieve the b68token
+    SBuf blob(proxy_auth);
+    if (!blob.isEmpty()) {
+        // trim prefix: WSP "Bearer" WSP
+        SBuf::size_type p = blob.findFirstNotOf(CharacterSet::WSP);
+        blob.consume(p);
+        p = blob.findFirstOf(CharacterSet::WSP);
+        blob.consume(p);
+        p = blob.findFirstNotOf(CharacterSet::WSP);
+        blob.consume(p);
+    }
+
+    // empty header? no auth details produced...
+    if (blob.isEmpty())
+        return auth_user_request;
+
+    // XXX: if the token contains non-b68token characters it is invalid.
+
+    Auth::User::Pointer auth_user;
+
+    // if there is a cached entry for this token use it
+    if (auto *entry = Auth::Bearer::Token::Cache.get(blob)) {
+        auth_user = (*entry)->user;
+
+    } else {
+        // generate a User object for this token and cache it
+        auto *usr = new Auth::Bearer::User(this, aRequestRealm);
+        usr->token = new Token(blob);
```

Comment:
Done.

Per your questions. They are separate because they do not have a 1:1 relationship, and tokens may need entries in the `TokenCache` long after their initial `User` object no longer references them.